app/models/ems_refresh/save_inventory_helper.rb

```diff
@@ -93,12 +93,19 @@ def save_inventory_with_findkey(association, hash, deletes, new_records, record_
       found = association.build(hash.except(:id))
       new_records << found
     else
-      found.update_attributes!(hash.except(:id, :type))
+      update_attributes!(found, hash, [:id, :type])
       deletes.delete(found) unless deletes.blank?
     end
     found
   end
 
+  def update_attributes!(ar_model, attributes, remove_keys)
+    ar_model.assign_attributes(attributes.except(*remove_keys))
+    ar_model.with_transaction_returning_status do
```

Comment:
@blomquisg we need to do this without the transaction, otherwise it has no optimization effect

The nil transactions are what we are trying to remove with ar_model.changed?